.travis.yml

```diff
@@ -1,37 +1,41 @@
-language: go
-
 addons:
   hosts:
     - quic.clemente.io
 
-go:
-  - 1.x
-  - tip
+env:
+    global:
+        - GO111MODULE=on
 
 matrix:
-  allow_failures:
-    - go: tip
-  fast_finish: true
+  include:
+    - language: go
+      go: 1.11.x
+      env:
+        - TEST_ARGS="-mod=readonly"
+        - TEST_ARGS="-mod=vendor"
+    - language: go
+      go: 1.x
+    - language: go
+      go: tip
+      env:
+        - TEST_ARGS="-mod=readonly"
+        - TEST_ARGS="-mod=vendor"
+
+allow_failures:
+  - go: tip
+fast_finish: true
 
 before_install:
   # Decrypts a script that installs an authenticated cookie
   # for git to use when cloning from googlesource.com.
   # Bypasses "bandwidth limit exceeded" errors.
   # See github.com/golang/go/issues/12933
- - if [ "$TRAVIS_PULL_REQUEST" = "false" ]; then openssl aes-256-cbc -K $encrypted_3df18f9af81d_key -iv $encrypted_3df18f9af81d_iv -in dist/gitcookie.sh.enc -out dist/gitcookie.sh -d; fi
+  - if [ "$TRAVIS_PULL_REQUEST" = "false" ]; then openssl aes-256-cbc -K $encrypted_3df18f9af81d_key -iv $encrypted_3df18f9af81d_iv -in dist/gitcookie.sh.enc -out dist/gitcookie.sh -d; fi
+  - curl -sfL https://install.goreleaser.com/github.com/golangci/golangci-lint.sh | bash -s -- -b $GOPATH/bin v1.10.2
 
 install:
   - if [ "$TRAVIS_PULL_REQUEST" = "false" ]; then bash dist/gitcookie.sh; fi
-  - go get -t ./...
-  - go get github.com/golang/lint/golint
-  - go get github.com/FiloSottile/vendorcheck
-  - go get github.com/alecthomas/gometalinter
 
 script:
-  - gometalinter --install
-  - gometalinter --disable-all -E vet -E gofmt -E misspell -E ineffassign -E goimports -E deadcode --tests --vendor ./...
-  - vendorcheck ./...
-  - go test -race ./...
-
-after_script:
-  - golint ./...
+  - golangci-lint run --deadline=10m --disable-all -E golint -E gofmt -E misspell -E ineffassign -E goimports -E deadcode ./...
+  - go test $TEST_ARGS -race ./...
```

Comment:
There should be two tests, one with `-mod=readonly` and one with `-mod=vendor`. Currently, it's only running once (probably with `-mod=vendor`, you don't know which one it's running)